scripts/upload_results_to_oep.py

```diff
@@ -0,0 +1,247 @@
+
+
+def write_metadata(metadata, schema, table, title, keywords):
+    metadata["name"] = table
+    metadata["title"] = title
+    metadata["PublicationDate"] = str(date.today())
+    # TODO: A method metadata.add_resource, add field would be handy
+    metadata["resources"][0]["name"] = f"{schema}.{table}"
+    metadata["keywords"] = keywords
+
+    return metadata
+
+
+if __name__ == "__main__":
+    filepath = pathlib.Path(sys.argv[1])
+    metadata_path = pathlib.Path(sys.argv[2])
+    logfile = sys.argv[3]
+
+    # make scenario name compatible with oep requirements
+    # (Names must consist of lowercase alpha-numeric words or underscores
+    # and start with a letter and must be have a maximumlength of 50)
+    scenario = "scenario_" + str(filepath.parts[1]).replace("-", "_").lower()
+
+    # set up the logger
+    logger = config.add_snake_logger(logfile, "upload_results_to_oep")
+
+    if not os.path.exists(metadata_path):
+        os.makedirs(metadata_path)
+
+    # define table names
+    def get_table_name(filename, scenario):
+        return f"{os.path.splitext(filename)[0]}_{scenario}"
+
+    # find data to upload
+    dict_table_filename = {
+        get_table_name(filename, scenario): filename
+        for filename in os.listdir(filepath)
+    }
+    logger.info(
+        "These files will be uploaded: " + ", ".join(dict_table_filename.values())
+    )
+
+    # To connect to the OEP you need your OEP Token and user name. Note: You ca view your token
+    # on your OEP profile page after
+    # [logging in](https://openenergy-platform.org/user/login/?next=/).
+    # The following command will prompt you for your token and store it as an environment variable.
+    # When you paste it here, it will only show dots instead of the actual string.
+    # save user name & token in environment as OEP_TOKEN & OEP_USER
+    db = oem2orm.setup_db_connection()
+
+    # Create the metadata and save it
+    for table, filename in dict_table_filename.items():
+        data_upload_df = pd.read_csv(
+            os.path.join(filepath, filename), encoding="utf8", sep=";", index_col=0
+        )
+
+        metadata = create_metadata(data_upload_df)
+
+        metadata = write_metadata(
+            metadata,
+            SCHEMA,
+            table,
+            f"Model results oemof-B3 {table}",
+            ["RLI", "oemof_b3"],
+        )
+
+        save_dict_to_json(metadata, metadata_path / f"{table}.json")
+
+        logger.info(f"Saved metadata to: {metadata_path}")
+
+    # Creating sql tables from oemetadata
+    # The next command will set up the tables. The collect_tables-function collects all metadata
+    # files in a folder, creates the SQLAlchemy ORM objects and returns them. The tables are
+    # ordered by foreign key. Having a valid metadata strings is necessary for the following steps.
+    tables_orm = oem2orm.collect_tables_from_oem(db, metadata_path)
+
+    # create tables
+    oem2orm.create_tables(db, tables_orm)
```

Comment:
@jh-RLI: Can we use the same function, but for single tables? This would allow us to not leave the loop here.

Reply:
What I mean: Do not do this: https://github.com/OpenEnergyPlatform/oem2orm/blob/80dffe65318c9ffa11c9167fa7930cf15929b133/oem2orm/oep_oedialect_oem2orm.py#L109
and this within the function https://github.com/OpenEnergyPlatform/oem2orm/blob/80dffe65318c9ffa11c9167fa7930cf15929b133/oem2orm/oep_oedialect_oem2orm.py#L298-L302

Reply:
There were too many tasks again today - if you can work with the [single_element] list, that would be great for now. I will update this as soon as I have more time to work on it. 
I created a issue here: https://github.com/OpenEnergyPlatform/oem2orm/issues/35

Reply:
Ok, I added an inline comment to remind us.